src/m365/pa/commands/app/app-permission-ensure.spec.ts

```diff
@@ -0,0 +1,324 @@
+  let commandInfo: CommandInfo;
+
+  const validEnvironmentName = 'Default-6a2903af-9c03-4c02-a50b-e7419599925b';
+  const validAppName = '784670e6-199a-4993-ae13-4b6747a0cd5d';
+  const validUserId = 'd2481133-e3ed-4add-836d-6e200969dd03';
+  const validUserName = 'john.doe@contoso.com';
+  const validGroupId = 'c6c4b4e0-cd72-4d64-8ec2-cfbd0388ec16';
+  const validGroupName = 'CLI Group';
+  const validRoleName = 'CanEdit';
+  const tenantId = '174290ec-373f-4d4c-89ea-9801dad0acd9';
+  const appPermissionEnsureResponse = {
+    put: [
+      {
+        id: "/providers/Microsoft.PowerApps/scopes/admin/apps/784670e6-199a-4993-ae13-4b6747a0cd5/permissions/241adbf6-2a56-4c72-81f2-69e75de6ac34",
+        properties: {
+          roleName: "CanView",
+          scope: "/providers/Microsoft.PowerApps/apps/784670e6-199a-4993-ae13-4b6747a0cd5",
+          principal: {
+            id: "d2481133-e3ed-4add-836d-6e200969dd03",
+            type: "User"
+          },
+          resourceResponses: [
+            {
+              id: "/providers/Microsoft.PowerApps/apps/784670e6-199a-4993-ae13-4b6747a0cd5",
+              statusCode: "Created",
+              responseCode: "ResourceShared",
+              message: "This was shared with 'CanView' permission.",
+              type: "/providers/Microsoft.PowerApps/apps"
+            }
+          ]
+        }
+      }
+    ]
+  };
+  const groupResponse = {
+    "id": validGroupId,
+    "deletedDateTime": null,
+    "classification": null,
+    "createdDateTime": "2017-11-29T03:27:05Z",
+    "description": "This is the Contoso Finance Group. Please come here and check out the latest news, posts, files, and more.",
+    "displayName": "Finance",
+    "groupTypes": [
+      "Unified"
+    ],
+    "mail": "finance@contoso.onmicrosoft.com",
+    "mailEnabled": true,
+    "mailNickname": "finance",
+    "onPremisesLastSyncDateTime": null,
+    "onPremisesProvisioningErrors": [],
+    "onPremisesSecurityIdentifier": null,
+    "onPremisesSyncEnabled": null,
+    "preferredDataLocation": null,
+    "proxyAddresses": [
+      "SMTP:finance@contoso.onmicrosoft.com"
+    ],
+    "renewedDateTime": "2017-11-29T03:27:05Z",
+    "securityEnabled": false,
+    "visibility": "Public"
+  };
+
+  before(() => {
+    cli = Cli.getInstance();
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.post,
+      cli.getSettingWithDefaultValue,
```

Comment:
Let's remove this.

Reply:
Like mentioned in the other PR, we have to use this since we use option sets in this command. So i've added `sinon.stub(cli, 'getSettingWithDefaultValue').callsFake(((settingName, defaultValue) => defaultValue));`